### ISSUE TITLE
kwm-overlay flashes when command-tab is used with mouse cursor

### ISSUE THREAD

**Issue by EdenSG** [koekeishiya/kwm#184]
Every time the mouse moves the overlay turns on and then off immediately after, resulting in flashing.

I very rarely use the mouse with the application switcher, so low priority.

Thanks!

**Comment by chrishoage** [comment]
Not really the same issue, but related to cmd-tab: If focus is given to a window by _any_ other method than kwm the overlay does not update. Using `cmd + tab` and cmd + ` give focus to other windows, but the overlay stays on the previous window

**Comment by koekeishiya** [comment]
>If focus is given to a window by any other method than kwm the overlay does not update. Using cmd + tab and cmd + ` give focus to other windows, but the overlay stays on the previous window

Fixed in latest commit

> Every time the mouse moves the overlay turns on and then off immediately after, resulting in flashing.

Pretty sure this was a side-effect after fixing a more annoying bug. Will look into it at some point.

**Comment by chrishoage** [comment]
It is fixed when focus is changed between the same display. Still broken when focus goes between displays (`cmd + tab` to an app that is on another display).

**Comment by koekeishiya** [comment]
> It is fixed when focus is changed between the same display. Still broken when focus goes between displays (cmd + tab to an app that is on another display). 

Hmm, not sure how to fix that. I use the "didActivateApplication notification" from OSX to handle the case.

**Comment by chrishoage** [comment]
Actually, I am a little bit wrong: It _does_ work between displays but only if if you leave the application switcher open for a second. 

Quickly cmd tabbing doesn't work, "slowly" cmd tabbing does. Strange.

**Comment by koekeishiya** [comment]
I noticed that the FocusWindowOfOSX function checked the wrong window list as well, just pushed a fix for that. I do not know if that will impact the behaviour you mentioned though.

**Comment by chrishoage** [comment]
> I do not know if that will impact the behaviour you mentioned though.

Just pulled, it actually fixed it

**Comment by justin-parus** [comment]
I still have the issue with "quick" cmd + tab not shifting the focus overlay to the next window. On same screen and desktop. Is homebrew bottle updated simultaneously as the git?

**Comment by chrishoage** [comment]
>  Is homebrew bottle updated simultaneously as the git?

kwm uses a distributed binary with homebrew, so it has to be updated manually. It looks like there is a pull request out to update the Formula to 1.1.3 right now that hasn't been merged yet.

**Comment by justin-parus** [comment]
Still understanding homebrew. So, I'm assuming once the change is merged and I update with homebrew I should have the fix? Anyway thanks, I don't mean to detract from the main discussion.

**Comment by chrishoage** [comment]
> So, I'm assuming once the change is merged and I update with homebrew I should have the fix?

Yup, you can also download it from the releases and replace the binaries that are in homebrew, or manually update the formula and update that way.

**Comment by justin-parus** [comment]
Thanks man! Appreciate the help.

**Comment by dominiklohmann** [comment]
Regarding the homebrew-binary PR, don't really know when they're gonna merge it. There's been no response since I opened it 2 days ago, which seems weird to me.

**Comment by chrishoage** [comment]
> Regarding the homebrew-binary PR, don't really know when they're gonna merge it. There's been no response since I opened it 2 days ago, which seems weird to me.

There is nothing stopping us from moving the Formula into to this repo and having people install it that way. 

`brew install koekeishiya/kwm/kwm`

**Comment by koekeishiya** [comment]
Figured I should probably explain the issue at hand, as we sort of sidetracked and left it for dead.

The reason it appears to be constantly flashing as the mouse is moved, is because of how Focus-Follows-Mouse works. In short, what is happening, is that the `focus of Kwm` is cleared (I do not recall why this happens, but there was a reason for it), and as the mouse is moved, it tries to re-focus the window behind the cmd+tab interface.

So to tackle this problem, we need some background as to how the internal representation of all this is within OSX. To keep it simple, everything regarding the Aqua Interface is more or less controlled by either the `Dock` or `Finder` application. 

Whenever the cmd+tab interface appears, or the dock itself is on screen (it does not matter how), the Dock on the internal side, is a window that takes up the entire screen. Meaning it is as if there was an invisible window stretching from (0,0) to (width, height) of your screen. 

To allow for Focus-Follows-Mouse to work for users that have the `Dock` visible, Kwm basically skips this window in the list of "windows allowed to gain focus", and that is the reason for this issue.
If Kwm does not skip this window, it prevents Focus-Follows-Mouse to work on machines that do not use "auto-hide dock"

**Comment by koekeishiya** [comment]
Fixed on master.
